docs/resource_sets.rst

```diff
@@ -0,0 +1,90 @@
+The changes will be pushed to the server when this smaller model is recompiled and exported in partial mode, but all other resource sets won't be impacted.
+
+While the remainder of this document will focus on the straightforward scenario of manually trimming down the model to facilitate quicker compilations,
+the partial compile feature is actually most useful in conjunction with additional tooling (such as a model generator based on a YAML file) or an Inmanta extension (such as ``LSM``) that offers dynamic entity construction.
+
+
+Resource sets
+###########################
+
+Instances of the ``std::ResourceSet`` entity serve as the model's representation of resource sets. The name of the set and a list of its resources are held by this object.
+These ``ResourceSet`` instances are found by the default exporter to ascertain which resources belong to which set.
+
+In the example below, 1000 networks of 5 routers each are created. Each router is part of its network's resource set.
+
+.. literalinclude:: resource_sets/basic_example_full.cf
+    :language: inmanta
+    :caption: main.cf
+
+
+Partial compiles
+###########################
+
+When a model is partially compiled, it only includes the entities and resources for the resource sets that need to be changed (as well as their dependencies on additional resources that aren't part of a resource set).
+It is the server's responsibility to create a new version of the model utilizing the resources from the old version and those from the partial compile.
+
+Only the resource sets that are present in the partially built model will be replaced when a partial export
+to the server is performed. Other sets' resources won't be impacted in any way.
+Shared resources are those that aren't a part of any resource collection and can always be added.
+
+
+The resources from the prior example would be updated by a partial export for the model below:
+
+.. literalinclude:: resource_sets/basic_example_partial.cf
+    :language: inmanta
+    :caption: main.cf
+
+As a result, network 0 would be changed to only have one router (the other resources are removed), but the other networks
+would continue to function as they had before (because their resource set was not present in the partial export).
+The comparable complete model would seem as follows:
+
+.. literalinclude:: resource_sets/basic_example_full_result.cf
+    :language: inmanta
+    :caption: main.cf
+
+Keep in mind that each resource set contains a collection of separate resources.
+Since the router instances for other sets do not exist at compilation time, it would be impossible to enforce a router index that was based just on the id and excluded the network.
+The liability of ensuring that resource sets are consistently defined for a set of resources that is distinct (is not part of the partial compile) from other sets remains with the model developer.
```

Comment:
hmmm, I have never come across an example that used this term in a negative way. All are to convey and depict authority. Anyway, it is just a word, if you do not feel like it is the best choice here, it can be changed.

Reply:
The sentence has changed. this is no longer valid.

Reply:
It was not necessarily that, I really was just wondering. I'm not a native English speaker so I wasn't sure. I wanted to raise the issue but if you are confident this would not carry such a connotation I have no objections.

Reply:
Sadly, I am not a native speaker either and am prone to many mistakes 😄 , In fact, I have practiced this language for less than 7 years now. Anyway, I had to change that line. I think it is a tad bit better now.